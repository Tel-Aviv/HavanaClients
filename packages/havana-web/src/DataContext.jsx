import React from 'react';

const userName = '';
export const DataContext = React.createContext(userName);