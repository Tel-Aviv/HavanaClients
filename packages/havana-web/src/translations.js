const translations = {
    en: {
        translation: {
            "Switch": "English",
            "product_name": "Attendance system",
            "to": " to day ",
            "cancel": "Cancel",
            "sure": "Are you sure?",
            "title": "Hours Report",
            "home": "Home",
            "settings": "Settings",
            "signature": "Signature",
            "stamp": "Stamp",
            "setting_title": "Your Pesrsonal Info",
            "details": "Info",
            "notifications": "Notifications",
            "plain": "Plain",
            "calendar": "Calendar",
            "yearly": "Yearly",
            "no_entry": "no entry",
            "no_exit": "no exit",
            "submit": "Submit",
            "approve": "Approve",
            "for_month": "for month",
            "miising_in": "missing entry",
            "missing_out": "missing exit",
            "in": "In",
            "out": "Out",
            "notes": "Notes",
            "notes_for_report": "Notes for Report",
            "missing_notes": "Please provide notes",
            "not_approved": "This report is not approves yet",
            "was_approved": "This report was approved at ",
            "not_submitted": "This report was not submitted yet",
            "print": "Print",
            "continue": "Continue",
            "upload": "Upload",
            "abs_docs": "Absense Docs",
            "save": "Save",
            "save_tooltip": "Save interim",
            "saved": "Your changes were saved",
            "hours": "hours",
            "months": "months",
            "send_to_approval": "Are you sure?",
            "validate": "validate",
            "close": "close",
            "invalid_items": "Invalid items",
            "validate_report": "Validate report",
            "report_validated": "Report is valid",
            "print_report": "Print Report",
            "add_record": "Add record",
            "edit_record": "Edit hourly",
            "add_entry_error": "Entry time is mandatory",
            "add_exit_error": "Exit time is mandatory",            
            "add_notes_error": "Please provide your notes",
            "move_to": "Forward",
            "exit_before_entry": "Exit before entry?",
            "manual_tag": "Has been edited manually",
            "summary": "Summary attendance for ",
            "printed_when": "Printed on ",
            "approved_when": "Approved on ",
            "approvals_list": "Reports assigned to you",
            "pending_reports": "Pending",
            "approved_reports": "Approved",
            "m1": "January",
            "m2": "Ferbuary",
            "m3": "March",
            "m4": "April",
            "m5": "May",
            "m6": "Juny",
            "m7": "July",
            "m8": "August",
            "m9": "September",
            "m10": "October",
            "m11": "November",
            "m12": "December",
            "reports_to": "Reports to",
            "no_direct_manager": "Could not find the direct manager. Please select one from the list nearby",
            "reject": "Reject",
            "approve_tooltip" : "Accept with possible comments",
            "reject_tooltip": "Reject and provide the reason",
            "forward_tooltip": "Move the approval to the next management level",
            "rejected_note": "This report was rejected by the manager with note: ",
            "rejected_reports": "Rejected",
            "approval_status": "Pending appoval",
            "resubmitted_status": "Re-submitted",
            "resubmitted_header": "This report was resubmitted",
            "report_code": "Report Type",
            "no_managers": "No managers were defined for this user",
            "support_page": "Support Page for Report",
            "expand": "Expand",
            "collapse": "Collapse",
            "refresh": "Refresh",
            "cancel_edit": "Sure to discard changes?",
            "why_add_record": "Free text - the essence of the activity",
            "report_full_day": "Report whole day",
            "job_description": "Job Description",
            "job_description_tooltip": "This text will be attached to a daily report",
            "job_description_error": "What do you did on this day?"
        }
    },
    he: {
        translation: {
            "Switch": "Hebrew",
            "product_name": "מערכת נוכחות",
            "to_day": " ליום ",
            "cancel": "ביטול",
            "sure": "האם ברצונך לבטל את השינויים?",
            "title": "דוח שעות",
            "home": "דף ראשי",
            "settings": "הגדרות אישיות",
            "signature": "חתימה",
            "stamp": "חותמת",
            "setting_title": "הגדרות אישיות",
            "details": "פרטים",
            "notifications": "דוחות לטיפולך",
            "plain": "טבלה",
            "calendar": "לוח חודשי",
            "yearly": "לוח שנתי",
            "no_entry": "כניסה חסרה",
            "no_exit": "יציאה חסרה",
            "submit": "העבר לאישור",
            "approve": "אישור",
            "for_month": " לחודש",
            "miising_in": "חסרה כניסה",
            "missing_out": "חסרה יציאה",
            "in": "כניסה",
            "out": "יציאה",
            "notes": "הערות",
            "notes_for_report": "הערות לדוח",
            "missing_notes": "אנא הזן הערות!",
            "not_approved": "דוח זה ממתין לאישור",
            "was_approved": "הדוח הזה אושר בתקריך",
            "not_submitted": "דוח זה לא נשלח לאישור",  
            "print": "הדפס",
            "continue": "המשך",
            "upload": "העלאה",
            "abs_docs": "מסמכי היעדרות",
            "save": "שמור",
            "save_tooltip": "שמור זמנית",
            "saved": "השינויים נשמרו",
            "hours": "שעות",
            "months": "חודשים",
            "send_to_approval": "הדוח שלך יישלח לאישור על ידי מנהל הישיר. לא ייתאפשרו עדכונים בדוח לפני האישור. לשלוח?",
            "validate": "בדוק",
            "close": "סגור",
            "invalid_items": "דיווחים לא תקינים",
            "validate_report": "אמת נתונים",
            "report_validated": "נתוני הדוח תקינים",
            "print_report": "הדפסת דוח",
            "add_record": "הוספת דיווח",
            "edit_record": "דיווח שעתי",
            "add_entry_error": "שכחת זמן כניסה?",
            "add_exit_error": "שכחת זמן יציאה?",            
            "add_notes_error": "בבקשה להוסיף הערות",
            "move_to": "העבר",
            "exit_before_entry": "יציאה לפני הכניסה?",
            "manual_tag": "נערך באופן ידני",
            "summary": "גיליון נוכחות מסכם לחודש",
            "printed_when": " הופק בתאריך ",
            "approved_when": "אושר בתאריך",
            "approvals_list": "דוחות בטיפולך",
            "pending_reports": "ממתינים",
            "approved_reports": "אושרו",
            "m1": "ינואר",
            "m2": "פברואר",
            "m3": "מרץ",
            "m4": "אפריל",
            "m5": "מאי",
            "m6": "יוני",
            "m7": "יולי",
            "m8": "אוגוסט",
            "m9": "ספטמבר",
            "m10": "אוקטובר",
            "m11": "נובמבר",
            "m12": "דצמבר",
            "reports_to": "מדווח ל",
            "no_direct_manager": "לא נמצא מנהל ישיר לאישור דוחות. נא בחר מנהל ישיר מהרשימה ליד הכפתור.",
            "reject": "דחה",
            "approve_tooltip" : "לאשר עם הערות אפשריות",
            "reject_tooltip": "דחה את הדוח והסבר מדוע",
            "forward_tooltip" : "העבר את האישור לדרג ניהול הבא",
            "rejected_note": "הדוח הזה נדחה על ידי המנהל עם ההערה: ",
            "rejected_reports": "נדחו",
            "approval_status": "ממתין לאישור",
            "resubmitted_status": "נשלח מחדש",
            "resubmitted_header": "דוח זה נשלח לאישור מחדש לאחר טיפול בהערת מנהל",
            "report_code": "קוד דיווח",
            "no_managers": "לא הוגדרו מנהלים עבור משתמש זה",
            "support_page": "דף תמיכה לדוח",
            "expand": "הרחב",
            "collapse": "צמצם",
            "refresh": "רענן",
            "cancel_edit": "האם ברצונך לבטל את השינויים ?",
            "why_add_record": "טקסט חופשי - מהות הפעילות",
            "report_full_day": "דיווח יום מלא",
            "job_description": "פירוט העבודה",
            "job_description_tooltip": "הפירוט ייצורף לדוח יומי",
            "job_description_error": "מה עשית ביום זה?"    
        }
    }
}    

export default translations;
