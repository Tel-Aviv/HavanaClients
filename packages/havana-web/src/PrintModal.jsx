import React, { useState, useEffect, useContext, useRef, Children } from 'react';
import { Modal, Button, Row, Col } from 'antd';
import ReactToPrint from 'react-to-print';
import moment from 'moment';
import { useTranslation } from "react-i18next";
import uniqid from 'uniqid'; 

import { DataContext } from './DataContext';

const PrintModal = (props) => {

    const [visible, setVisible] = useState(false)
    const dataContext = useContext(DataContext);
    const componentRef = useRef();
    const { t } = useTranslation();

    useEffect( () => {
        setVisible(!visible)
    }, [props.visible])

    const gettTitle = () => (
        <div style={{ margin: '0 auto' }}>
            {t('print_report')}
        </div>
    )

    const onCancel = () => {
        setVisible(false);
        props.closed & props.closed();
    }

    return (
        <Modal title={gettTitle()}
            width='70%'
            closable={true}
            forceRender={true}
            visible={visible}
            onCancel={onCancel}
            footer={[
                <ReactToPrint key={uniqid()} 
                            copyStyles={true}
                            removeAfterPrint={true}
                            trigger={() => <Button type="primary">{t('print')}</Button>}
                            content={() => componentRef.current}
                />,
                <Button key={uniqid()} onClick={onCancel}>{t('cancel')}</Button>
            ]}>
                <div ref={componentRef} style={{textAlign: 'center'}} className='pdf-container'>
                    <div className='pdf-title'>{dataContext.user.name}</div>
                    <div className='pdf-title'>{t('summary')} {props.month}/{props.year}</div>
                    {
                        props.children
                    }
                    <Row>
                        <Col span={9} style={{
                                display: 'flex',
                                justifyContent: 'flex-end'
                            }}>
                            {
                                props.signature ? 
                                <img className='footer-signature' src={props.signature} /> :
                                null
                            }
                        </Col>
                        <Col span={3}>
                            <div className='footer-print'>{t('signature')}</div>        
                        </Col>
                        <Col span={6}>
                            <div className='footer-print'>סה"כ { props.totals } שעות</div>
                        </Col>
                        <Col span={6}>
                            <div className='footer-print'>{t('printed_when')} {moment().format('DD/MM/YYYY')}</div>
                        </Col>                                                 
                    </Row>
                </div>
        </Modal>
    )
}

export default PrintModal;