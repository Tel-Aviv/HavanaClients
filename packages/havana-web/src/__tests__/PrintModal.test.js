import './matchMedia.mock';

import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import i18n from 'i18next';
import { initReactI18next } from "react-i18next";
import translations from '../translations';

import { DataContext } from '../DataContext';
import PrintModal from '../PrintModal';

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources: translations,
    lng: "he",
    debug: false
  })

describe('PrintModal', () => {

    const context = {
        user: 'אולג קליימן',
    }

    const cancelFn = jest.fn();

    const printModalComponent = 
        <DataContext.Provider value={context}>
          <PrintModal visible={true}
              closed={cancelFn}
              month={10}
              year={2020}
              signature={null}
              totals={160.37}>
                <div>Dummy</div>
          </PrintModal>
        </DataContext.Provider>

    // const printModalComponent = <ReportContext.Provider value={ {
    //                                 codes: []
    //                             }
    //                             }>
    //         <AddRecordModal />
    //     </ReportContext.Provider>

    beforeAll( () => {
        jest.resetModules()
    })     
    test('Layout Snap', () => {

        // const component = renderer.create(printModalComponent);

        // let tree = component.toJSON();
        // expect(tree).toMatchSnapshot();

    })

    // test('Test functionality (@testing-library)', () => {
    //     render(printModalComponent)
        
    //     const printButtonElement = screen.getByLabelText('print');
    //     expect(printButtonElement).toBeInTheDocument();
    //     fireEvent(printButtonElement, new MouseEvent('click', {
    //         bubbles: true,
    //         cancelable: true,
    //     }));        
    // })
})