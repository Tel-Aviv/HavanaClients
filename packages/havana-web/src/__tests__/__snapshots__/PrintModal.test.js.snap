// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintModal Layout PrintModal Snap 1`] = `null`;
